index.js

```diff
@@ -33,51 +44,107 @@ const Request = {
     const key = Verifier.manifestHash(manifest)
     const id = HypercoreID.normalize(key)
 
+    const isDrive = state.start !== state.end && c.uint8.decode(state) !== 0
+    const blobs = isDrive
+      ? {
+          length: c.uint.decode(state),
+          treeHash: c.fixed32.decode(state)
+        }
+      : null
+
     return {
       version,
       id,
       key,
       length,
       fork,
       treeHash,
-      manifest
+      manifest,
+      isDrive,
+      blobs
     }
   }
 }
 
 module.exports = {
-  async generate (core, { length = core.length, fork = core.fork, manifest = null } = {}) {
-    if (!core.opened) await core.ready()
+  generate,
+  generateDrive,
+  decode,
+  signable
+}
 
-    if (core.core.compat && !manifest) throw new Error('Cannot generate signing requests for compat cores')
-    if (core.fork !== fork) throw new Error('Core should have the same fork')
-    if (!manifest) manifest = core.manifest
+async function generate (core, { length = core.length, fork = core.fork, manifest = null } = {}) {
+  if (!core.opened) await core.ready()
 
-    return c.encode(Request, {
-      version: VERSION,
-      length,
-      fork,
-      treeHash: await core.treeHash(length),
-      manifest
-    })
-  },
-  decode (buffer) {
-    const state = { start: 0, end: buffer.byteLength, buffer }
-    const req = Request.decode(state)
+  if (core.blobs) return generateDrive(core, { length, fork, manifest })
 
-    if (req.length === 0) throw new Error('Refusing to sign length = 0')
-    if (state.start < state.end) throw new Error('Unparsed padding left in request, bailing')
+  if (core.core.compat && !manifest) throw new Error('Cannot generate signing requests for compat cores')
+  if (core.fork !== fork) throw new Error('Core should have the same fork')
+  if (!manifest) manifest = core.manifest
 
-    return req
-  },
-  signable (pub, req) {
-    const v = req.manifest.version
-    for (const s of req.manifest.signers) {
-      if (s.publicKey.equals(pub)) {
-        return caps.treeSignable(v === 0 ? s.namespace : req.key, req.treeHash, req.length, req.fork)
+  return c.encode(Request, {
+    version: VERSION,
+    length,
+    fork,
+    treeHash: await core.treeHash(length),
+    manifest,
+    blobs: null
+  })
+}
+
+async function generateDrive (drive, { length = drive.core.length, fork = drive.core.fork, manifest = null }) {
+  if (drive.core.core.compat && !manifest) throw new Error('Cannot generate signing requests for compat cores')
```

Comment:
Edge case, but is it possible to have a Hyperdrive with a non-compat db core, but a compat blobs core? And if so: is that an issue? (Might need to error on that too)

Reply:
no they move together, nice q tho